[GTK][WPE] Enable Cookie Store API by default

clopez opened the pull request:
#### 24324208dd4d73cadc1dfd11fc0e09ef746d4109
<pre>
[GTK][WPE] Enable Cookie Store API by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=292416">https://bugs.webkit.org/show_bug.cgi?id=292416</a>

Reviewed by Rupin Mittal and Michael Catanzaro.

Cookie Store API &lt;<a href="https://wicg.github.io/cookie-store">https://wicg.github.io/cookie-store</a>&gt;
is a Web spec that is currently supported by the three
main Web engines.

Before 289626@main it was enabled by default for the
Apple ports, and after that it depends on a build-time
conditional macro that this patch defaults to enable
for GTK and WPE.

I tested to enable it by default for the GTK and WPE
ports and I have ran the WPT tests (not the imported
ones here in webkit, but the real WPT tests) and most
of the tests pass now.

There are no changes to the layout test expectations
with this patch because WTR runs by default with all
the run-time settings enabled, so layout tests were
already running with this enabled.

* Source/WTF/wtf/PlatformEnableGlib.h:

Canonical link: <a href="https://commits.webkit.org/294448@main">https://commits.webkit.org/294448@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/47076c958c4510d54aacb0797507d0e477927507

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101741 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21409 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11725 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106899 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52375 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21717 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29911 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77463 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34490 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104748 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16776 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91879 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57801 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16604 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9897 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51726 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/94413 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86462 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9974 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109292 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/100351 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28874 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21262 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86473 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29235 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88080 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86044 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30770 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8492 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23037 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16568 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28802 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34092 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/124012 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28613 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34449 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31936 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30172 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Let's try this and see what breaks.

RupinMittal reviewed (approved):
LGTM! It's nice to see Cookie Store support being added for GTK and WPE :)

I'm approving--though someone from glib should probably also take a look.

clopez committed:
[GTK][WPE] Enable Cookie Store API by default
https://bugs.webkit.org/show_bug.cgi?id=292416

Reviewed by Rupin Mittal and Michael Catanzaro.

Cookie Store API <https://wicg.github.io/cookie-store>
is a Web spec that is currently supported by the three
main Web engines.

Before 289626@main it was enabled by default for the
Apple ports, and after that it depends on a build-time
conditional macro that this patch defaults to enable
for GTK and WPE.

I tested to enable it by default for the GTK and WPE
ports and I have ran the WPT tests (not the imported
ones here in webkit, but the real WPT tests) and most
of the tests pass now.

There are no changes to the layout test expectations
with this patch because WTR runs by default with all
the run-time settings enabled, so layout tests were
already running with this enabled.

* Source/WTF/wtf/PlatformEnableGlib.h:

Canonical link: https://commits.webkit.org/294448@main
